<a href="https://colab.research.google.com/github/anishchettla/Time-series-data-analysis-with-ARIMA-model/blob/main/Financial_Management.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>



#**Calculate the Hurdle rate /WACC for a listed firm**

#**INDUSTRY: GEMS & JEWELLERY**





# **Calculations of regression beta of compaines in Gems and Jewellery Industry**


- **Darshan Orna Limited = dol**

- **Radhika JewelTech Limited = rjl**

- **Thangmayil Jewellery Limited = tjl**

- **Goldiam International Limited = gil** 







In [ ]:
#Install required libraries
!pip install yfinance
import matplotlib.pyplot as plt
import statsmodels.api as sm
import yfinance as yf
import pandas as pd
import numpy as np
import seaborn as sns
import scipy.stats as stats
warnings.filterwarnings("ignore")

     |████████████████████████████████| 63 kB 1.3 MB/s 
     |████████████████████████████████| 6.4 MB 19.6 MB/s 
  Attempting uninstall: requests
    Found existing installation: requests 2.23.0
    Uninstalling requests-2.23.0:
      Successfully uninstalled requests-2.23.0
  Attempting uninstall: lxml
    Found existing installation: lxml 4.2.6
    Uninstalling lxml-4.2.6:
      Successfully uninstalled lxml-4.2.6
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
google-colab 1.0.0 requires requests~=2.23.0, but you have requests 2.27.1 which is incompatible.
datascience 0.10.6 requires folium==0.2.1, but you have folium 0.8.3 which is incompatible.


/usr/local/lib/python3.7/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


NameError: ignored

*`Historical Data of Darshan Orna Limited.`*

In [ ]:
#Download stock prices for the security monthly data last 5 years
dol_df = yf.download('DARSHANORNA.BO', 
                      start='2017-04-01', 
                      end='2021-03-31', 
                      progress=False,interval ='1mo'
)
dol_df.head()

Open       High        Low      Close  Adj Close     Volume
Date                                                                        
2017-04-01  16.976191  24.428572  16.976191  24.428572  24.382305    50400.0
2017-05-01  24.428572  29.952379  23.976191  28.571428  28.517315  2520000.0
2017-06-01  28.571428  28.571428  28.571428  28.571428  28.517315    16800.0
2017-07-01  28.571428  39.523808  27.428572  39.500000  39.425190   260400.0
2017-08-01  39.523808  45.238094  39.523808  42.380951  42.300686   126000.0

*`Historical Data of Radhika JewelTech Limited.`*

In [ ]:
#Download stock prices for the security monthly data last 5 years
rjl_df = yf.download('RADHIKAJWE.BO', 
                      start='2017-04-01', 
                      end='2021-03-31', 
                      progress=False,interval ='1mo'
)
rjl_df.head()

Open       High   Low      Close  Adj Close     Volume
Date                                                                   
2017-04-01  12.500000  24.500000  12.5  18.000000  17.163448  1177600.0
2017-05-01  18.000000  18.750000  13.0  14.000000  13.349349  1331200.0
2017-06-01  13.800000  21.120001  13.0  20.639999  19.680754   873600.0
2017-07-01  21.299999  25.750000  17.6  22.200001  21.168255   779200.0
2017-08-01  22.750000  22.750000  16.6  17.600000  16.782040   160000.0

*`Historical Data of Thangamayil Jewellery Limited`*

In [ ]:
#Download stock prices for the security monthly data last 5 years
tjl_df = yf.download('THANGAMAYL.NS', 
                      start='2017-04-01', 
                      end='2021-03-31', 
                      progress=False,interval ='1mo'
)
tjl_df.head()

Open        High         Low       Close   Adj Close  \
Date                                                                     
2017-04-01  228.149994  270.000000  228.100006  265.399994  240.692902   
2017-05-01  265.399994  272.049988  215.050003  242.899994  220.287506   
2017-06-01  248.300003  268.799988  229.199997  243.199997  220.559540   
2017-07-01  250.000000  348.100006  230.000000  322.649994  292.613190   
2017-07-18         NaN         NaN         NaN         NaN         NaN   

               Volume  
Date                   
2017-04-01   195929.0  
2017-05-01   181212.0  
2017-06-01   114946.0  
2017-07-01  2209741.0  
2017-07-18        NaN

*`Historical Data of Goldiam International Limited`*

In [ ]:
#Download stock prices for the security monthly data last 5 years
gil_df = yf.download('GOLDIAM.NS', 
                      start='2017-04-01', 
                      end='2021-03-31', 
                      progress=False,interval ='1mo'
)
gil_df.head()

Open       High    Low      Close  Adj Close     Volume
Date                                                                    
2017-04-01  17.020000  19.959999  17.02  18.750000  14.918503  4198430.0
2017-05-01  18.750000  19.299999  16.40  16.750000  13.327196  3550550.0
2017-06-01  16.980000  17.799999  16.08  16.110001  12.817975  1124275.0
2017-07-01  16.450001  18.059999  16.01  16.870001  13.422673  2183740.0
2017-08-01  17.030001  17.760000  14.16  16.549999  13.168062  1805375.0

*`Historical Data of NIFTY`*

In [ ]:
#Download Market index closing prices
nifty_df = yf.download('^BSESN',start='2017-04-01', 
                      end='2021-03-31', 
                      progress=False,interval ='1mo',
)
nifty_df.head()


Open          High           Low         Close  \
Date                                                                 
2017-04-01  29737.730469  30184.220703  29241.480469  29918.400391   
2017-05-01  30021.490234  31255.279297  29804.119141  31145.800781   
2017-06-01  31117.089844  31522.869141  30680.660156  30921.609375   
2017-07-01  31156.039062  32672.660156  31017.109375  32514.939453   
2017-08-01  32579.800781  32686.480469  31128.019531  31730.490234   

               Adj Close  Volume  
Date                              
2017-04-01  29918.400391  194700  
2017-05-01  31145.800781  282400  
2017-06-01  30921.609375  329400  
2017-07-01  32514.939453  258300  
2017-08-01  31730.490234  241300

*`NIFTY Returns`*

In [ ]:
#Calculate Log returns
nifty_df['Return_NIFTY'] = np.log(nifty_df['Adj Close']/nifty_df['Adj Close'].shift(1))
print(nifty_df['Return_NIFTY']) 

Date
2017-04-01         NaN
2017-05-01    0.040206
2017-06-01   -0.007224
2017-07-01    0.050244
2017-08-01   -0.024422
2017-09-01   -0.014180
2017-10-01    0.059847
2017-11-01   -0.001922
2017-12-01    0.027007
2018-01-01    0.054516
2018-02-01   -0.050788
2018-03-01   -0.036201
2018-04-01    0.064361
2018-05-01    0.004597
2018-06-01    0.002858
2018-07-01    0.059804
2018-08-01    0.027240
2018-09-01   -0.064611
2018-10-01   -0.050530
2018-11-01    0.049623
2018-12-01   -0.003487
2019-01-01    0.005209
2019-02-01   -0.010794
2019-03-01    0.075309
2019-04-01    0.009231
2019-05-01    0.017338
2019-06-01   -0.008079
2019-07-01   -0.049792
2019-08-01   -0.003965
2019-09-01    0.035123
2019-10-01    0.037106
2019-11-01    0.016430
2019-12-01    0.011211
2020-01-01   -0.012937
2020-02-01   -0.061426
2020-03-01   -0.262058
2020-04-01    0.134699
2020-05-01   -0.039119
2020-06-01    0.074038
2020-07-01    0.074248
2020-08-01    0.026798
2020-09-01   -0.014613
2020-10-01    0.039812
2020-1

In [ ]:
nifty_df1 = nifty_df.dropna(axis=0)
nifty_df1.head()

Open          High           Low         Close  \
Date                                                                 
2017-05-01  30021.490234  31255.279297  29804.119141  31145.800781   
2017-06-01  31117.089844  31522.869141  30680.660156  30921.609375   
2017-07-01  31156.039062  32672.660156  31017.109375  32514.939453   
2017-08-01  32579.800781  32686.480469  31128.019531  31730.490234   
2017-09-01  31769.339844  32524.109375  31081.830078  31283.720703   

               Adj Close  Volume  Return_NIFTY  
Date                                            
2017-05-01  31145.800781  282400      0.040206  
2017-06-01  30921.609375  329400     -0.007224  
2017-07-01  32514.939453  258300      0.050244  
2017-08-01  31730.490234  241300     -0.024422  
2017-09-01  31283.720703  233900     -0.014180

In [ ]:
nifty_df1_returns = nifty_df1['Return_NIFTY']
nifty_df1_returns.head()

Date
2017-05-01    0.040206
2017-06-01   -0.007224
2017-07-01    0.050244
2017-08-01   -0.024422
2017-09-01   -0.014180
Name: Return_NIFTY, dtype: float64

*`Mean Market Return calculation for CAPM model`*

In [ ]:
returns_nifty = nifty_df1_returns
returns_nifty.describe()

count    47.000000
mean      0.010717
std       0.060515
min      -0.262058
25%      -0.013558
50%       0.009231
75%       0.049934
max       0.134699
Name: Return_NIFTY, dtype: float64

*Mean Market Return(Rm_Mean) is 1.0717%*

*`Darshan Orna Limited Returns`*

In [ ]:
#Caclulate log returns
dol_df['Return_DOL'] = np.log(dol_df['Adj Close']/dol_df['Adj Close'].shift(1))
print(dol_df['Return_DOL']) 

Date
2017-04-01         NaN
2017-05-01    0.156654
2017-06-01    0.000000
2017-07-01    0.323894
2017-08-01    0.070398
2017-09-01    0.155754
2017-10-01   -0.028281
2017-11-01    0.296545
2017-12-01    0.162519
2018-01-01    0.213093
2018-02-01   -0.084260
2018-03-01   -0.802177
2018-04-01    0.083994
2018-05-01   -0.031476
2018-06-01   -0.097639
2018-07-01    0.258412
2018-08-01    0.213193
2018-08-20         NaN
2018-09-01         NaN
2018-09-19         NaN
2018-10-01         NaN
2018-11-01   -0.035091
2018-12-01    0.417888
2019-01-01    0.271682
2019-02-01   -1.005014
2019-03-01   -0.275920
2019-04-01   -0.156842
2019-05-01    0.125718
2019-06-01    0.260699
2019-07-01   -0.048598
2019-08-01    0.161240
2019-09-01    0.195032
2019-10-01   -0.079415
2019-11-01   -0.002043
2019-12-01    0.040086
2020-01-01    0.021381
2020-02-01   -0.384532
2020-03-01    0.044206
2020-04-01   -0.064178
2020-05-01   -0.119225
2020-06-01   -0.212647
2020-07-01   -0.445409
2020-08-01   -0.349091
2020-0

In [ ]:
dol_df1 = dol_df.dropna(axis=0)
dol_df1.head()

Open       High        Low      Close  Adj Close     Volume  \
Date                                                                           
2017-05-01  24.428572  29.952379  23.976191  28.571428  28.517315  2520000.0   
2017-06-01  28.571428  28.571428  28.571428  28.571428  28.517315    16800.0   
2017-07-01  28.571428  39.523808  27.428572  39.500000  39.425190   260400.0   
2017-08-01  39.523808  45.238094  39.523808  42.380951  42.300686   126000.0   
2017-09-01  40.714287  49.523808  40.285713  49.523808  49.430012   134400.0   

            Return_DOL  
Date                    
2017-05-01    0.156654  
2017-06-01    0.000000  
2017-07-01    0.323894  
2017-08-01    0.070398  
2017-09-01    0.155754

In [ ]:
dol_df1_returns = dol_df1['Return_DOL']
dol_df1_returns.head()

Date
2017-05-01    0.156654
2017-06-01    0.000000
2017-07-01    0.323894
2017-08-01    0.070398
2017-09-01    0.155754
Name: Return_DOL, dtype: float64

In [ ]:
returns_dol = pd.merge(nifty_df1_returns,dol_df1_returns,how='inner',on='Date')
returns_dol.head()
returns_dol.describe()

Return_NIFTY  Return_DOL
count     45.000000   45.000000
mean       0.013752   -0.011401
std        0.060041    0.313082
min       -0.262058   -1.005014
25%       -0.010794   -0.097639
50%        0.011211    0.000000
75%        0.050244    0.161240
max        0.134699    0.938270

*`Radhika JewelTech Limited Returns`*

In [ ]:
#Caclulate log returns
rjl_df['Return_RJL'] = np.log(rjl_df['Adj Close']/rjl_df['Adj Close'].shift(1))
print(rjl_df['Return_RJL'])

Date
2017-04-01         NaN
2017-05-01   -0.251314
2017-06-01    0.388174
2017-07-01    0.072861
2017-08-01   -0.232193
2017-09-01   -0.061513
2017-10-01    0.055815
2017-11-01    0.366625
2017-12-01    0.488398
2018-01-01   -0.092883
2018-02-01   -0.223144
2018-03-01   -0.053049
2018-04-01    0.036242
2018-05-01   -0.006803
2018-06-01   -0.304894
2018-07-01    0.018349
2018-08-01   -0.046520
2018-09-01   -0.113329
2018-10-01    0.041782
2018-11-01   -0.007702
2018-12-01   -0.002581
2019-01-01   -0.050342
2019-02-01   -0.108990
2019-03-01    0.058840
2019-04-01    0.158224
2019-05-01    0.218254
2019-06-01   -0.103184
2019-07-01   -0.245122
2019-07-22         NaN
2019-08-01         NaN
2019-09-01    0.029532
2019-10-01    0.002642
2019-11-01   -0.024033
2019-12-01    0.010753
2020-01-01   -0.077817
2020-02-01   -0.059541
2020-03-01    0.006116
2020-04-01    0.015129
2020-05-01    0.150282
2020-06-01   -0.086307
2020-07-01   -0.043172
2020-08-01   -0.092373
2020-09-01   -0.042840
2020-1

In [ ]:
rjl_df1 = rjl_df.dropna(axis=0)
rjl_df1.head()

Open       High   Low      Close  Adj Close     Volume  \
Date                                                                      
2017-05-01  18.000000  18.750000  13.0  14.000000  13.349349  1331200.0   
2017-06-01  13.800000  21.120001  13.0  20.639999  19.680754   873600.0   
2017-07-01  21.299999  25.750000  17.6  22.200001  21.168255   779200.0   
2017-08-01  22.750000  22.750000  16.6  17.600000  16.782040   160000.0   
2017-09-01  17.750000  22.750000  16.5  16.549999  15.780837   428800.0   

            Return_RJL  
Date                    
2017-05-01   -0.251314  
2017-06-01    0.388174  
2017-07-01    0.072861  
2017-08-01   -0.232193  
2017-09-01   -0.061513

In [ ]:
rjl_df1_returns = rjl_df1['Return_RJL']
rjl_df1_returns.head()

Date
2017-05-01   -0.251314
2017-06-01    0.388174
2017-07-01    0.072861
2017-08-01   -0.232193
2017-09-01   -0.061513
Name: Return_RJL, dtype: float64

In [ ]:
returns_rjl = pd.merge(nifty_df1_returns,rjl_df1_returns,how='inner',on='Date')
returns_rjl.head()
returns_rjl.describe()

Return_NIFTY  Return_RJL
count     46.000000   46.000000
mean       0.011036   -0.004310
std        0.061144    0.155676
min       -0.262058   -0.304894
25%       -0.013869   -0.090857
50%        0.010221   -0.007252
75%        0.050089    0.052306
max        0.134699    0.488398

*`Thangmayil Jewellery Limited Returns`*


In [ ]:
#Caclulate log returns
tjl_df['Return_TJL'] = np.log(tjl_df['Adj Close']/tjl_df['Adj Close'].shift(1))
print(tjl_df['Return_TJL'])

Date
2017-04-01         NaN
2017-05-01   -0.088588
2017-06-01    0.001234
2017-07-01    0.282684
2017-07-18         NaN
2017-08-01         NaN
2017-09-01    0.061503
2017-10-01    0.110362
2017-11-01    0.309373
2017-12-01    0.091476
2018-01-01   -0.060893
2018-02-01   -0.145698
2018-03-01   -0.008365
2018-04-01    0.104284
2018-05-01   -0.063339
2018-06-01   -0.134178
2018-07-01   -0.062404
2018-07-24         NaN
2018-07-25         NaN
2018-08-01         NaN
2018-09-01   -0.269512
2018-10-01    0.270510
2018-11-01   -0.129818
2018-12-01    0.074351
2019-01-01   -0.060542
2019-02-01   -0.125844
2019-03-01    0.056462
2019-04-01   -0.049503
2019-05-01   -0.024409
2019-06-01    0.101720
2019-07-01   -0.173000
2019-07-24         NaN
2019-07-25         NaN
2019-08-01         NaN
2019-09-01    0.097148
2019-10-01    0.096277
2019-11-01   -0.021244
2019-12-01    0.001573
2020-01-01    0.000000
2020-02-01    0.253519
2020-03-01   -0.642753
2020-03-23         NaN
2020-04-01         NaN
2020-0

In [ ]:
tjl_df1 = tjl_df.dropna(axis=0)
tjl_df1.head()

Open        High         Low       Close   Adj Close  \
Date                                                                     
2017-05-01  265.399994  272.049988  215.050003  242.899994  220.287506   
2017-06-01  248.300003  268.799988  229.199997  243.199997  220.559540   
2017-07-01  250.000000  348.100006  230.000000  322.649994  292.613190   
2017-09-01  355.000000  388.899994  335.000000  362.950012  331.720123   
2017-10-01  362.950012  466.950012  355.149994  405.299988  370.426147   

               Volume  Return_TJL  
Date                               
2017-05-01   181212.0   -0.088588  
2017-06-01   114946.0    0.001234  
2017-07-01  2209741.0    0.282684  
2017-09-01    76754.0    0.061503  
2017-10-01   170337.0    0.110362

In [ ]:
tjl_df1_returns = tjl_df1['Return_TJL']
tjl_df1_returns.head()

Date
2017-05-01   -0.088588
2017-06-01    0.001234
2017-07-01    0.282684
2017-09-01    0.061503
2017-10-01    0.110362
Name: Return_TJL, dtype: float64

In [ ]:
returns_tjl = pd.merge(nifty_df1_returns,tjl_df1_returns,how='inner',on='Date')
returns_tjl.head()
returns_tjl.describe()

Return_NIFTY  Return_TJL
count     43.000000   43.000000
mean       0.008608    0.016030
std        0.059994    0.168200
min       -0.262058   -0.642753
25%       -0.013558   -0.061649
50%        0.009231    0.001573
75%        0.049934    0.099434
max        0.108402    0.309373

*`Goldiam International Limited Returns`*

In [ ]:
#Caclulate log returns
gil_df['Return_GIL'] = np.log(gil_df['Adj Close']/gil_df['Adj Close'].shift(1))
print(gil_df['Return_GIL'])

Date
2017-04-01         NaN
2017-05-01   -0.112795
2017-06-01   -0.038958
2017-07-01    0.046097
2017-08-01   -0.019151
2017-09-01   -0.052725
2017-10-01   -0.007673
2017-11-01    0.043949
2017-12-01    0.035600
2018-01-01   -0.095725
2018-02-01   -0.082219
2018-03-01   -0.178646
2018-04-01    0.405747
2018-05-01   -0.254382
2018-06-01   -0.205291
2018-07-01    0.092137
2018-08-01    0.226561
2018-09-01   -0.098239
2018-09-12         NaN
2018-10-01         NaN
2018-11-01    0.204032
2018-12-01   -0.099289
2019-01-01   -0.046785
2019-02-01    0.061500
2019-02-21         NaN
2019-03-01         NaN
2019-04-01    0.213470
2019-05-01    0.192673
2019-06-01    0.120690
2019-07-01   -0.327704
2019-08-01    0.062975
2019-08-20         NaN
2019-09-01         NaN
2019-09-12         NaN
2019-09-13         NaN
2019-09-19         NaN
2019-10-01         NaN
2019-11-01   -0.069896
2019-12-01    0.096260
2020-01-01    0.034218
2020-02-01    0.059685
2020-02-24         NaN
2020-03-01         NaN
2020-0

In [ ]:
gil_df1 = gil_df.dropna(axis=0)
gil_df1.head()

Open       High    Low      Close  Adj Close     Volume  \
Date                                                                       
2017-05-01  18.750000  19.299999  16.40  16.750000  13.327196  3550550.0   
2017-06-01  16.980000  17.799999  16.08  16.110001  12.817975  1124275.0   
2017-07-01  16.450001  18.059999  16.01  16.870001  13.422673  2183740.0   
2017-08-01  17.030001  17.760000  14.16  16.549999  13.168062  1805375.0   
2017-09-01  16.420000  19.350000  15.46  15.700000  12.491761  4431680.0   

            Return_GIL  
Date                    
2017-05-01   -0.112795  
2017-06-01   -0.038958  
2017-07-01    0.046097  
2017-08-01   -0.019151  
2017-09-01   -0.052725

In [ ]:
gil_df1_returns = gil_df1['Return_GIL']
gil_df1_returns.head()

Date
2017-05-01   -0.112795
2017-06-01   -0.038958
2017-07-01    0.046097
2017-08-01   -0.019151
2017-09-01   -0.052725
Name: Return_GIL, dtype: float64

In [ ]:
returns_gil = pd.merge(nifty_df1_returns,gil_df1_returns,how='inner',on='Date')
returns_gil.head()
returns_gil.describe()

Return_NIFTY  Return_GIL
count     39.000000   39.000000
mean       0.013903    0.024775
std        0.045596    0.156579
min       -0.064611   -0.327704
25%       -0.013558   -0.076058
50%        0.005209    0.035600
75%        0.049934    0.089655
max        0.134699    0.452450

*`Darshan Orna Limited regression beta calculations`*

In [ ]:
x = returns_dol['Return_NIFTY']
m = returns_dol['Return_DOL']
x_sm = sm.add_constant(x)

/usr/local/lib/python3.7/dist-packages/statsmodels/tsa/tsatools.py:117: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


Assumptions of OLS regression
A linear relationship between the dependent and independent variables - The linear regression model is linear in parameters
The independent variables are not highly correlated with each other - no multicollinearity
The variance of the residuals is constant - no heteroskedasticity
Random sampling of observation and no autocorrelation
Assumption of Normality of Errors

Ri = αi + βi(RM) + εi.........Market model

• Ri = Return of security i

• αi = The return from the asset that is not related to the market’s return. This is “alpha” return from the security

• βi = Beta or the return from the security explained by the market index’s return

• RM = The market index’s return

• εi = Error term for past returns not explained by the regression equation

In [ ]:
#Regression - Market model 
model_dol = sm.OLS(m,x_sm)

OLS estimation or Ordinary Least square Minimizes the sum of squared errors

In [ ]:
results_dol = model_dol.fit()

In [ ]:
#Regression result
results_dol.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:             Return_DOL   R-squared:                       0.003
Model:                            OLS   Adj. R-squared:                 -0.020
Method:                 Least Squares   F-statistic:                    0.1364
Date:                Thu, 21 Apr 2022   Prob (F-statistic):              0.714
Time:                        16:51:41   Log-Likelihood:                -11.017
No. Observations:                  45   AIC:                             26.03
Df Residuals:                      43   BIC:                             29.65
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const           -0.0154      0.048     -0.319      0.751      -0.113       0.082
Return_NIFTY     0.2932      0.794      0.369      0.714      -1.308       1.894
==============================================================================
Omnibus:                        9.602   Durbin-Watson:                   1.793
Prob(Omnibus):                  0.008   Jarque-Bera (JB):               15.171
Skew:                          -0.506   Prob(JB):                     0.000508
Kurtosis:                       5.659   Cond. No.                         16.8
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

#*Regression beta for Darshan Orna Limited stock = 0.293 and significant at 5%.*


**` Radhika JewelTech Limited regression beta calculations`**

In [ ]:
x = returns_rjl['Return_NIFTY']
n = returns_rjl['Return_RJL']
x_sm = sm.add_constant(x)

/usr/local/lib/python3.7/dist-packages/statsmodels/tsa/tsatools.py:117: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


In [ ]:
#Regression - Market model 
model_rjl = sm.OLS(n,x_sm)

In [ ]:
results_rjl = model_rjl.fit()

In [ ]:
#Regression result
results_rjl.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:             Return_RJL   R-squared:                       0.008
Model:                            OLS   Adj. R-squared:                 -0.015
Method:                 Least Squares   F-statistic:                    0.3487
Date:                Thu, 21 Apr 2022   Prob (F-statistic):              0.558
Time:                        16:51:51   Log-Likelihood:                 20.975
No. Observations:                  46   AIC:                            -37.95
Df Residuals:                      44   BIC:                            -34.29
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const           -0.0068      0.024     -0.289      0.774      -0.054       0.041
Return_NIFTY     0.2258      0.382      0.591      0.558      -0.545       0.996
==============================================================================
Omnibus:                       13.265   Durbin-Watson:                   1.722
Prob(Omnibus):                  0.001   Jarque-Bera (JB):               15.242
Skew:                           1.027   Prob(JB):                     0.000490
Kurtosis:                       4.933   Cond. No.                         16.5
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

#*Regression beta for Radhika JewelTech Limited stock = 0.226 and significant at 5%.*

*`Thangamayil Jewellery Limited regression beta calculations`*

In [ ]:
x = returns_tjl['Return_NIFTY']
O = returns_tjl['Return_TJL']
x_sm = sm.add_constant(x)

/usr/local/lib/python3.7/dist-packages/statsmodels/tsa/tsatools.py:117: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


In [ ]:
#Regression - Market model 
model_tjl = sm.OLS(O,x_sm)

In [ ]:
results_tjl = model_tjl.fit()

In [ ]:
#Regression result
results_tjl.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:             Return_TJL   R-squared:                       0.349
Model:                            OLS   Adj. R-squared:                  0.333
Method:                 Least Squares   F-statistic:                     21.99
Date:                Thu, 21 Apr 2022   Prob (F-statistic):           3.03e-05
Time:                        16:51:55   Log-Likelihood:                 25.375
No. Observations:                  43   AIC:                            -46.75
Df Residuals:                      41   BIC:                            -43.23
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const            0.0018      0.021      0.084      0.934      -0.041       0.045
Return_NIFTY     1.6565      0.353      4.689      0.000       0.943       2.370
==============================================================================
Omnibus:                        7.548   Durbin-Watson:                   2.246
Prob(Omnibus):                  0.023   Jarque-Bera (JB):                6.424
Skew:                           0.887   Prob(JB):                       0.0403
Kurtosis:                       3.660   Cond. No.                         16.9
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

#*Regression beta for Thangamayil Jewellery Limited stock = 1.656 and significant at 5%.*

*`Goldiam International Limited regression beta calculations`*

In [ ]:
x = returns_gil['Return_NIFTY']
p = returns_gil['Return_GIL']
x_sm = sm.add_constant(x)

/usr/local/lib/python3.7/dist-packages/statsmodels/tsa/tsatools.py:117: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


In [ ]:
#Regression - Market model 
model_gil = sm.OLS(p,x_sm)

In [ ]:
results_gil = model_gil.fit()

In [ ]:
#Regression result
results_gil.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:             Return_GIL   R-squared:                       0.143
Model:                            OLS   Adj. R-squared:                  0.120
Method:                 Least Squares   F-statistic:                     6.193
Date:                Thu, 21 Apr 2022   Prob (F-statistic):             0.0174
Time:                        16:52:08   Log-Likelihood:                 20.500
No. Observations:                  39   AIC:                            -37.00
Df Residuals:                      37   BIC:                            -33.67
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const            0.0067      0.025      0.272      0.787      -0.043       0.057
Return_NIFTY     1.3004      0.523      2.489      0.017       0.242       2.359
==============================================================================
Omnibus:                        0.972   Durbin-Watson:                   2.122
Prob(Omnibus):                  0.615   Jarque-Bera (JB):                0.490
Skew:                           0.270   Prob(JB):                        0.783
Kurtosis:                       3.099   Cond. No.                         22.2
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

#*Regression beta for Goldiam International Limited stock = 1.300 and significant at 5%.*

#**Hurdle rate /WACC calculations**

#*Tax Rate, Risk free rate(YTM) and Market Risk Mean(calculated from above) on 31st March, 2021 are 25%, 6.177% and 1.0717% respectively.*

In [ ]:
#Rm_Mean, Rf and T hold Risk free rate and Tax Rate respectively
T = 0.25
Rf = 0.06177
Rm_Mean = 0.010717
print("Tax Rate on 31st March, 2021 "+str(T*100)+"%")
print("Risk free rate "+str(Rf*100)+"%")
print( "Market Risk Mean(calculated from above) "+str(Rm_Mean*100)+"%")

Tax Rate on 31st March, 2021 25.0%
Risk free rate 6.177%
Market Risk Mean(calculated from above) 1.0717%


#*co_beta_regeression stores regression beta value of a company and*
#*co_credit_default_spread holds credit default spread of a company*
*(Synthetic Rating and Credit Default Spread for companies are taken as per " http://pages.stern.nyu.edu/~adamodar/New_Home_Page/valquestions/syntrating.htm ")*

#-  *Darshan Orna Limited = dol (AAA rated and 0.75% spread).*

#-  *Radhika JewelTech Limited = rjl (AAA rated and 0.75% spread).*

#-  *Thangmayil Jewellery Limited = tjl (AA rated and 0.75% spread).*

#-  *Goldiam International Limited = gil (AAA rated and 0.75% spread).*






In [ ]:
dol_beta_regression = 0.293
dol_credit_default_spread = 0.0075
rjl_beta_regression = 0.226
rjl_credit_default_spread = 0.0075
tjl_beta_regression = 1.656
tjl_credit_default_spread = 0.01
gil_beta_regression = 1.300
gil_credit_default_spread = 0.0075

*`This function is used to calculate company's data like Market Cap, D/E, Ws, Wb and Interest Coverage Ratio.`*

In [ ]:
def data_calculations(co_data, co_beta_regression, co_credit_default_spread):
  co_data.append(co_data[1]*co_data[2])
  co_data.append(co_data[3]/co_data[6])
  co_data.append(co_data[7]+1)
  co_data.append(1/co_data[8])
  co_data.append(co_data[7]/co_data[8])
  co_data.append(0)
  co_data.append(co_beta_regression)
  co_data.append(co_data[4]/co_data[5])
  co_data.append(co_credit_default_spread)
  print("Below is the data of "+str(co_data[0])+" on 31st March, 2021")
  print("Share Price of ", co_data[0],"Rs."+str(co_data[1]))
  print("No. of Outstanding Shares of", co_data[0], co_data[2])
  print("Market Capitalization of", co_data[0], "is Rs."+str(co_data[6]))
  print("Market value of debt of", co_data[0], "is Rs."+str(co_data[3]))
  print("EBIT of", co_data[0], "is "+str(co_data[4]))
  print("Interest Expense of", co_data[0], "is "+str(co_data[5]))
  print("Debt to Equity ratio(D/E) of", co_data[0], "is "+str(co_data[7]))
  print("Value of the firm to Equity Ratio(V/E) of", co_data[0], "is "+str(co_data[8]))
  print("Weight of Common Stock/Equity(Ws) of", co_data[0], "is "+str(co_data[9]))
  print("Weight of Debt instrument(Wb) of", co_data[0], "is "+str(co_data[10]))
  print("Weight of Preffered Stock(Wp) of", co_data[0], "is "+str(co_data[11]))
  print("Regression beta of", co_data[0], "is "+str(co_data[12]))
  print("Interest Coverage Ratio of", co_data[0], "is "+str(co_data[13]))
  print("Credit Default Spread of", co_data[0], "is "+str(co_data[14]))
  return co_data

*`This function is used to calculate cost of equity capital(Rs) top down or regression approach. `*



In [ ]:
def cost_of_equity_top_down(co_data, Rm, Rf):
  Rs_top_down = Rf+co_data[12]*(Rf-Rm)
  co_data.insert(15, Rs_top_down)
  print("Top-Down Cost of Equity Capital(Rs) of", co_data[0], "is "+str(co_data[15]*100)+"%")
  return co_data

*`This function is used to unlever the beta of a company. `*

In [ ]:
def unlever_beta(co_data, T):
  #Unlevering the beta of a company
  co_unlever_beta = co_data[12]/(1+((1-T)*co_data[7]))
  co_data.insert(16,co_unlever_beta)
  print("Unlevered beta of", co_data[0], "is "+str(co_data[16]))
  return co_data

*`This function is used to relever the beta of a company(co_data) through comparable co.s(co1_data, co2_data and co3_data). `*

In [ ]:
def relever_beta(co_data, co1_data, co2_data, co3_data, T):
  #Levering of beta through comparable co.s
  unlevered_beta_average = (co1_data[16]+co2_data[16]+co3_data[16])/3
  co_data.insert(17,unlevered_beta_average)
  relevered_beta = (unlevered_beta_average)*(1+(1-T)*co_data[7])
  co_data.insert(18, relevered_beta)
  print("Relevered beta of", co_data[0], "is "+str(co_data[18]))
  return co_data

*`This function is used to calculate cost of equity capital(Rs) bottom-up approach. `*

In [ ]:
def cost_of_equity_bottom_up(co_data, Rm, Rf):
  Rs_bottom_up = Rf+co_data[18]*(Rf-Rm)
  co_data.insert(19, Rs_bottom_up)
  print("Bottom-Up Cost of Equity Capital(Rs) of", co_data[0], "is "+str(co_data[19]*100)+"%")
  return co_data

*`This function is used to calculate Cost of Debt(Rb).`*

In [ ]:
def cost_of_debt(co_data, Rf):
  Rb = Rf + co_data[14]
  co_data.insert(20,Rb)
  print("Cost of Debt(Rb)", co_data[0], "is "+str(co_data[20]*100)+"%")
  return co_data

*`This function is used to calculate Weighted Average Cost of Capital(WACC).`*
*`(Cost of Equity Capital from bottom up approach is used)`*

In [ ]:
def cost_of_capital(co_data, T):
  WACC = (co_data[20]*co_data[10]*(1-T))+(co_data[19]*co_data[9])
  co_data.insert(21,WACC)
  print("Cost of Capital", co_data[0], "is "+str(co_data[21]*100)+"%")
  return co_data

#**Price, Outstanding Shares, Debt, EBIT and Interest Expense of on 31st March, 2021 from yahoo finance for**

- *Darshan Orna Limited are Rs.14.25, 23421420, Rs.2373000, Rs.20426000 and Rs.440290*

- *Radhika Jeweltech Limited are Rs.14.72, 7638100, Rs.340677000, Rs.337518000 and Rs.2978000*

- *Thangamayil Jewellery Limited are Rs.1147.50, 524340, Rs.4770104000, Rs.1367451000 and Rs.197736000*

- *Goldiam International Limited are Rs.147.90, 23073140, Rs.1326827000, Rs. 975005000, Rs.2987000*

#**respectively.**





*(Market value of debt = Book value of debt)*

*(Wp = 0 because there is no preferred stock for selected companies)*


In [ ]:
#list co_data contains data of a company from yahoo finance"
dol_data = ["Darshan Orna Limited", 14.25, 23421420, 2373000, 20426000, 440290]
rjl_data = ["Radhika Jeweltech Limited", 14.72, 7638100, 340677000, 337518000, 2978000]
tjl_data = ["Thangamayil Jewellery Limited", 1147.50, 524340, 4770104000, 1367451000, 197736000]
gil_data = ["Goldiam International Limited", 147.90, 23073140, 1326827000, 975005000, 2987000]

*`Calculations of company's data like Market Cap, D/E, Ws, Wb and Interest Coverage Ratio`*

In [ ]:
"""data_calculations function is called to calculate company's data like 
   Market Cap, D/E, Ws, Wb and Interest Coverage Ratio"""
dol_data = data_calculations(dol_data, dol_beta_regression, dol_credit_default_spread)

Below is the data of Darshan Orna Limited on 31st March, 2021
Share Price of  Darshan Orna Limited Rs.14.25
No. of Outstanding Shares of Darshan Orna Limited 23421420
Market Capitalization of Darshan Orna Limited is Rs.333755235.0
Market value of debt of Darshan Orna Limited is Rs.2373000
EBIT of Darshan Orna Limited is 20426000
Interest Expense of Darshan Orna Limited is 440290
Debt to Equity ratio(D/E) of Darshan Orna Limited is 0.007110000836391375
Value of the firm to Equity Ratio(V/E) of Darshan Orna Limited is 1.0071100008363914
Weight of Common Stock/Equity(Ws) of Darshan Orna Limited is 0.9929401943874189
Weight of Debt instrument(Wb) of Darshan Orna Limited is 0.007059805612581163
Weight of Preffered Stock(Wp) of Darshan Orna Limited is 0
Regression beta of Darshan Orna Limited is 0.293
Interest Coverage Ratio of Darshan Orna Limited is 46.392150627995186
Credit Default Spread of Darshan Orna Limited is 0.0075




#-  *Market Capitalization or Market Value of Equity of Darshan Orna Limited is Rs.333755235.*

#-  *Market value of debt or book value of debt of Darshan Orna Limited is Rs.2373000.*

#-  *Debt to Equity ratio(D/E) of Darshan Orna Limited is 0.007.*


#-  *Interest Coverage Ratio of Darshan Orna Limited is 46.392.*





In [ ]:
rjl_data = data_calculations(rjl_data, rjl_beta_regression, rjl_credit_default_spread)

Below is the data of Radhika Jeweltech Limited on 31st March, 2021
Share Price of  Radhika Jeweltech Limited Rs.14.72
No. of Outstanding Shares of Radhika Jeweltech Limited 7638100
Market Capitalization of Radhika Jeweltech Limited is Rs.112432832.0
Market value of debt of Radhika Jeweltech Limited is Rs.340677000
EBIT of Radhika Jeweltech Limited is 337518000
Interest Expense of Radhika Jeweltech Limited is 2978000
Debt to Equity ratio(D/E) of Radhika Jeweltech Limited is 3.0300490874409354
Value of the firm to Equity Ratio(V/E) of Radhika Jeweltech Limited is 4.030049087440935
Weight of Common Stock/Equity(Ws) of Radhika Jeweltech Limited is 0.24813593539501919
Weight of Debt instrument(Wb) of Radhika Jeweltech Limited is 0.7518640646049808
Weight of Preffered Stock(Wp) of Radhika Jeweltech Limited is 0
Regression beta of Radhika Jeweltech Limited is 0.226
Interest Coverage Ratio of Radhika Jeweltech Limited is 113.33713901947615
Credit Default Spread of Radhika Jeweltech Limited is 

#-  *Market Capitalization of Radhika Jeweltech Limited is Rs.112432832.*

#-  *Market value of debt of Radhika Jeweltech Limited is Rs.340677000.*

#-  *Debt to Equity ratio(D/E) of Radhika Jeweltech Limited is 3.030.*

#-  *Interest Coverage Ratio of Radhika Jeweltech Limited is 113.337.*

In [ ]:
tjl_data = data_calculations(tjl_data, tjl_beta_regression, tjl_credit_default_spread)

Below is the data of Thangamayil Jewellery Limited on 31st March, 2021
Share Price of  Thangamayil Jewellery Limited Rs.1147.5
No. of Outstanding Shares of Thangamayil Jewellery Limited 524340
Market Capitalization of Thangamayil Jewellery Limited is Rs.601680150.0
Market value of debt of Thangamayil Jewellery Limited is Rs.4770104000
EBIT of Thangamayil Jewellery Limited is 1367451000
Interest Expense of Thangamayil Jewellery Limited is 197736000
Debt to Equity ratio(D/E) of Thangamayil Jewellery Limited is 7.927973026864855
Value of the firm to Equity Ratio(V/E) of Thangamayil Jewellery Limited is 8.927973026864855
Weight of Common Stock/Equity(Ws) of Thangamayil Jewellery Limited is 0.11200750685412407
Weight of Debt instrument(Wb) of Thangamayil Jewellery Limited is 0.8879924931458759
Weight of Preffered Stock(Wp) of Thangamayil Jewellery Limited is 0
Regression beta of Thangamayil Jewellery Limited is 1.656
Interest Coverage Ratio of Thangamayil Jewellery Limited is 6.915538900351

#-   *Market Capitalization of Thangamayil Jewellery Limited is Rs.601680150.*

#-   *Market value of debt of Thangamayil Jewellery Limited is Rs.4770104000.*

#-   *Debt to Equity ratio(D/E) of Thangamayil Jewellery Limited is 7.928.*

#-    *Interest Coverage Ratio of Thangamayil Jewellery Limited is 6.916.*

In [ ]:
gil_data = data_calculations(gil_data, gil_beta_regression, gil_credit_default_spread)

Below is the data of Goldiam International Limited on 31st March, 2021
Share Price of  Goldiam International Limited Rs.147.9
No. of Outstanding Shares of Goldiam International Limited 23073140
Market Capitalization of Goldiam International Limited is Rs.3412517406.0
Market value of debt of Goldiam International Limited is Rs.1326827000
EBIT of Goldiam International Limited is 975005000
Interest Expense of Goldiam International Limited is 2987000
Debt to Equity ratio(D/E) of Goldiam International Limited is 0.388811789697286
Value of the firm to Equity Ratio(V/E) of Goldiam International Limited is 1.388811789697286
Weight of Common Stock/Equity(Ws) of Goldiam International Limited is 0.7200399704397427
Weight of Debt instrument(Wb) of Goldiam International Limited is 0.2799600295602573
Weight of Preffered Stock(Wp) of Goldiam International Limited is 0
Regression beta of Goldiam International Limited is 1.3
Interest Coverage Ratio of Goldiam International Limited is 326.41613659189824

#-  *Market Capitalization of Goldiam International Limited is Rs.3412517406.*

#-  *Market value of debt of Goldiam International Limited is is Rs.1326827000.*

#-  *Debt to Equity ratio(D/E) of Goldiam International Limited is 0.389.*

#-  *Interest Coverage Ratio of Goldiam International Limited is 326.416.*

*`Cost of Equity Capital(Rs) top down or regression approach calculations`*

In [ ]:
"""cost_of_equity_top_down function is called to calculate cost of equity 
   capital(Rs) top down or regression approach"""
dol_data = cost_of_equity_top_down(dol_data, Rm_Mean, Rf)
rjl_data = cost_of_equity_top_down(rjl_data, Rm_Mean, Rf)
tjl_data = cost_of_equity_top_down(tjl_data, Rm_Mean, Rf)
gil_data = cost_of_equity_top_down(gil_data, Rm_Mean, Rf)

Top-Down Cost of Equity Capital(Rs) of Darshan Orna Limited is 7.6728529000000005%
Top-Down Cost of Equity Capital(Rs) of Radhika Jeweltech Limited is 7.330797799999999%
Top-Down Cost of Equity Capital(Rs) of Thangamayil Jewellery Limited is 14.631376799999998%
Top-Down Cost of Equity Capital(Rs) of Goldiam International Limited is 12.81389%




#-   *Top-Down Cost of Equity Capital(Rs) of Darshan Orna Limited is 7.673%.*

#-   *Top-Down Cost of Equity Capital(Rs) of Radhika Jeweltech Limited is 7.331%.*

#-   *Top-Down Cost of Equity Capital(Rs) of Thangamayil Jewellery Limited is 14.6314%.*

#-   *Top-Down Cost of Equity Capital(Rs) of Goldiam International Limited is 12.814%.*







In [ ]:
#unlever_beta function is called to unlever the beta of a company
dol_data = unlever_beta(dol_data, T)
rjl_data = unlever_beta(rjl_data, T)
tjl_data = unlever_beta(tjl_data, T)
gil_data = unlever_beta(gil_data, T)

Unlevered beta of Darshan Orna Limited is 0.29144586474343354
Unlevered beta of Radhika Jeweltech Limited is 0.06905957449401437
Unlevered beta of Thangamayil Jewellery Limited is 0.23841129038654862
Unlevered beta of Goldiam International Limited is 1.0064966710140113


#-*Unlevered beta of Darshan Orna Limited is 0.291.*

#-*Unlevered beta of Radhika Jeweltech Limited is 0.069.*

#-*Unlevered beta of Thangamayil Jewellery Limited is 0.238.*

#-*Unlevered beta of Goldiam International Limited is 1.006.*

In [ ]:
"""relever_beta function is called to relever the beta of a company through 
  comparable co.s"""
dol_data = relever_beta(dol_data, rjl_data, tjl_data, gil_data, T)
rjl_data = relever_beta(rjl_data, dol_data, tjl_data, gil_data, T)
tjl_data = relever_beta(tjl_data, dol_data, rjl_data, gil_data, T)
gil_data = relever_beta(gil_data, dol_data, rjl_data, tjl_data, T)

Relevered beta of Darshan Orna Limited is 0.44032475620132516
Relevered beta of Radhika Jeweltech Limited is 1.675924819271497
Relevered beta of Thangamayil Jewellery Limited is 3.1650563345190466
Relevered beta of Goldiam International Limited is 0.25785538125585344


#-*Relevered beta of Darshan Orna Limited is 0.440.*

#-*Relevered beta of Radhika Jeweltech Limited is 1.676.*

#-*Relevered beta of Thangamayil Jewellery Limited is 3.165.*

#-*Relevered beta of Goldiam International Limited is 0.258.*

In [ ]:
"""cost_of_equity_bottom_up function is called to calculate cost of equity 
   capital(Rs) bottom-up approach"""
dol_data = cost_of_equity_bottom_up(dol_data, Rm_Mean, Rf)
rjl_data = cost_of_equity_bottom_up(rjl_data, Rm_Mean, Rf)
tjl_data = cost_of_equity_bottom_up(tjl_data, Rm_Mean, Rf)
gil_data = cost_of_equity_bottom_up(gil_data, Rm_Mean, Rf)

Bottom-Up Cost of Equity Capital(Rs) of Darshan Orna Limited is 8.424989977834626%
Bottom-Up Cost of Equity Capital(Rs) of Radhika Jeweltech Limited is 14.733098979826774%
Bottom-Up Cost of Equity Capital(Rs) of Thangamayil Jewellery Limited is 22.335562104620088%
Bottom-Up Cost of Equity Capital(Rs) of Goldiam International Limited is 7.4934290779255095%


#-   *Bottom-Up Cost of Equity Capital(Rs) of Darshan Orna Limited is 8.425%.*

#-   *Bottom-Up Cost of Equity Capital(Rs) of Radhika Jeweltech Limited is 14.733%.*

#-   *Bottom-Up Cost of Equity Capital(Rs) of Thangamayil Jewellery Limited is 22.336%.*

#-  *Bottom-Up Cost of Equity Capital(Rs) of Goldiam International Limited is 7.493%.*

In [ ]:
#cost_of_debt function is called to calculate cost of debt (Rb)
dol_data = cost_of_debt(dol_data, Rf)
rjl_data = cost_of_debt(rjl_data, Rf)
tjl_data = cost_of_debt(tjl_data, Rf)
gil_data = cost_of_debt(gil_data, Rf)

Cost of Debt(Rb) Darshan Orna Limited is 6.927%
Cost of Debt(Rb) Radhika Jeweltech Limited is 6.927%
Cost of Debt(Rb) Thangamayil Jewellery Limited is 7.177%
Cost of Debt(Rb) Goldiam International Limited is 6.927%


# -   *Cost of Debt(Rb) of Darshan Orna Limited is 6.927%.*

# -   *Cost of Debt(Rb) of Radhika Jeweltech Limited is 6.927%.*

# -   *Cost of Debt(Rb) of Thangamayil Jewellery Limited is 7.177%.*

# -   *Cost of Debt(Rb) of Goldiam International Limited is 6.927%.*

In [ ]:
#cost_of_capital function is used to calculate WACC
dol_data = cost_of_capital(dol_data, T)
rjl_data = cost_of_capital(rjl_data, T)
tjl_data = cost_of_capital(tjl_data, T)
gil_data = cost_of_capital(gil_data, T)

Cost of Capital Darshan Orna Limited is 8.40218864141193%
Cost of Capital Radhika Jeweltech Limited is 7.561933078265745%
Cost of Capital Thangamayil Jewellery Limited is 7.281592218004912%
Cost of Capital Goldiam International Limited is 6.850030795334719%


# -   *Cost of Capital/Hurdle Rate(WACC) of Darshan Orna Limited is 8.402%.*

# -   *Cost of Capital of Radhika Jeweltech Limited is 7.562%.*

# -   *Cost of Capital of Thangamayil Jewellery Limited is 7.282%.*

# -   *Cost of Capital of Goldiam International Limited is 6.850%.*

##References: " https://finance.yahoo.com/ "